### PR TITLE
Trivial rename: clarify test-only client.

### PR DESCRIPTION
**Pull request by jkh52**
Replace `cmd/client` with `cmd/test-client` throughout, and restore some lost docker make targets to verify (via `make docker-build-test`).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,8 +72,8 @@ build: bin/proxy-agent bin/proxy-server bin/proxy-test-client bin/http-test-serv
 bin/proxy-agent: proto/agent/agent_grpc.pb.go konnectivity-client/proto/client/client_grpc.pb.go bin cmd/agent/main.go
 	GO111MODULE=on go build -o bin/proxy-agent cmd/agent/main.go
 
-bin/proxy-test-client: konnectivity-client/proto/client/client_grpc.pb.go bin cmd/client/main.go
```
Comment (cheftako) [inline Makefile:75]: Switch to $(SOURCE)

Comment (jkh52) [reply Makefile]: Done (had to rebase to https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/482 anyway).

**Comment by jkh52** [comment]
Manual sanity checks:

```
$ make bin/proxy-test-client
$ make docker-build-test
```

**Pull request merged**